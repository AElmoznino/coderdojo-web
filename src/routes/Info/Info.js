// @flow

import React from 'react'
import type { Match } from 'react-router-dom'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import ReactMarkdown from 'react-markdown'
import GridColumn from 'components/Grid/GridColumn'
import GridSection from 'components/Grid/GridSection'
import LoadingIndicator from 'components/LoadingIndicator/LoadingIndicator'
import ErrorMessage from 'components/ErrorMessage/ErrorMessage'

const GET_PAGE = gql`
  query GetPage($pageId: String) {
    page(where: { pageId: $pageId }) {
      pageBody
    }
  }
`

export const Info = ({ match }: Match) => (
  <Query query={GET_PAGE} variables={{ pageId: match.params.pageId }}>
    {({ data: { page }, error, loading }) => {
      if (loading) return <LoadingIndicator />
      if (error) return <ErrorMessage />
      if (!page) return null

      return (
        <GridColumn>
          <GridSection>
            <ReactMarkdown>{page.pageBody}</ReactMarkdown>
          </GridSection>
        </GridColumn>
      )
    }}
  </Query>
)

export default Info
