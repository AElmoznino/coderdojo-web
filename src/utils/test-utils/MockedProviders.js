import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'
import ApolloClient from 'apollo-boost'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'

jest.unmock('react-apollo')
jest.unmock('react-router-dom')

type MockedProvidersProps = {
  children: React.Node,
  loading?: boolean,
  response?: Object,
}

const MockedProviders = ({
  children,
  loading,
  response = {},
}: MockedProvidersProps) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    uri: 'http://localhost:4000',
    fetch: mockedNetworkFetch,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedProviders
