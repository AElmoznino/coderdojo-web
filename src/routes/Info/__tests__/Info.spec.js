import React from 'react'
import { mount } from 'enzyme'
import Info from '../Info'
import MockedProviders from 'utils/test-utils/MockedProviders'

jest.unmock('graphql-tag')
jest.unmock('react-apollo')

describe('components/Info', () => {
  let component
  let match
  let mockedResponse

  beforeEach(() => {
    match = {
      params: {
        pageId: 'about',
      },
    }

    mockedResponse = {
      page: {
        __typename: 'Page',
        pageBody:
          '# Om CoderDojo Webbskolan\n\nWebbskolan är ett projekt framtaget av CoderDojo Stockholm för att lära barn och ungdomar att programmera för webben. Här finns ett urval av övningar i HTML, CSS och Javascript. Tanken är att man ska kunna jobba helt självgående.\n\nSidan är optimerad för Google Chrome och anpassad för bärbar dator då det är vad barnen/ungdomarna har med sig till CoderDojo.\n\nAll källkod finns tillgänglig på [GitHub](https://github.com/AElmoznino/coderdojo-web).\n\n## Om CoderDojo\nCoderDojo är en kodklubb för barn och ungdomar mellan 7 och 17 år. I skrivande stund (januari 2018) finns det i 111 länder. Det är gratis att gå med och alla volontärer arbetar ideellt, eftersom vi tycker att alla barn ska få chansen att upptäcka hur kul programmering är.\n        \nLäs mer om [CoderDojo](https://coderdojo.com/) och [CoderDojo Stockholm](http://coderdojostockholm.se), där du även kan se vårt schema för kommande dojos .',
      },
    }

    component = mount(
      <MockedProviders response={mockedResponse}>
        <Info match={match} />
      </MockedProviders>,
    )
  })

  it('renders Info', done => {
    setImmediate(() => {
      component.update()

      expect(component).toMatchSnapshot()

      done()
    })
  })
})
