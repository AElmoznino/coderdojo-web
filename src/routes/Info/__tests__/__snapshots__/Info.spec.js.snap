// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Info renders Info 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa cBlAJt"
  >
    <styled.div>
      <div
        className="sc-bwzfXH LKvyz"
      >
        Array [
          <h1>
            Om CoderDojo Webbskolan
          </h1>,
          <p
            key="paragraph-3-1"
          >
            Webbskolan är ett projekt framtaget av CoderDojo Stockholm för att lära barn och ungdomar att programmera för webben. Här finns ett urval av övningar i HTML, CSS och Javascript. Tanken är att man ska kunna jobba helt självgående.
          </p>,
          <p
            key="paragraph-5-1"
          >
            Sidan är optimerad för Google Chrome och anpassad för bärbar dator då det är vad barnen/ungdomarna har med sig till CoderDojo.
          </p>,
          <p
            key="paragraph-7-1"
          >
            All källkod finns tillgänglig på 
            <a
              href="https://github.com/AElmoznino/coderdojo-web"
              key="link-7-34"
            >
              GitHub
            </a>
            .
          </p>,
          <h2>
            Om CoderDojo
          </h2>,
          <p
            key="paragraph-10-1"
          >
            CoderDojo är en kodklubb för barn och ungdomar mellan 7 och 17 år. I skrivande stund (januari 2018) finns det i 111 länder. Det är gratis att gå med och alla volontärer arbetar ideellt, eftersom vi tycker att alla barn ska få chansen att upptäcka hur kul programmering är.
          </p>,
          <p
            key="paragraph-12-1"
          >
            Läs mer om 
            <a
              href="https://coderdojo.com/"
              key="link-12-12"
            >
              CoderDojo
            </a>
             och 
            <a
              href="http://coderdojostockholm.se"
              key="link-12-52"
            >
              CoderDojo Stockholm
            </a>
            , där du även kan se vårt schema för kommande dojos .
          </p>,
        ]
      </div>
    </styled.div>
  </div>
</styled.div>
`;
